packages/date-parser/src/result.js

```diff
@@ -0,0 +1,79 @@
+import { merge } from 'lodash';
+import luxonFromChronoStruct from './helpers/luxonFromChronoStruct';
+
+export default class Result {
+  /**
+   * @param {Date} ref
+   * @param {Number} index
+   * @param {String} text
+   * @param {Chrono.ParsingComponents} start
+   * @param {Chrono.ParsingComponents} end
+   * @param {String} weekStartDay
+   */
+  constructor ({
+    ref, index, text, start, end, weekStartDay,
+  }) {
+    this.ref = ref;
+    this.index = index;
+    this.text = text;
+    this.start = start;
+    this.end = end;
+    this.weekStartDay = weekStartDay;
+  }
+
+  toObject () {
+    return {
+      ref: this.ref,
+      index: this.index,
+      text: this.text,
+      start: this.start,
+      end: this.end,
+      weekStartDay: this.weekStartDay,
+    };
+  }
+
+  asLuxon () {
+    return merge(
+      this.toObject(), {
+        start: this.start ? luxonFromChronoStruct(this.start) : null,
+        end: this.end ? luxonFromChronoStruct(this.end) : null,
+      },
+    );
+  }
+
+  // Note that the luxon instance contains the original timezone
+  // The date here is formatted from that instance, so it's still implicit in that timezone
+  asDate () {
+    const luxonResult = this.asLuxon();
+    return merge(
+      luxonResult, {
+        start: luxonResult.start ? luxonResult.start.toFormat('YYYY-MM-DD') : null,
+        end: luxonResult.end ? luxonResult.end.toFormat('YYYY-MM-DD') : null,
+      },
+    );
+  }
+
+  // Convert the timezone to UTC then return the values, hence,
+  // the original timezone offset is lost
+  asTimestampUtc () {
+    const luxonResult = this.asLuxon();
+    return merge(
+      luxonResult, {
+        start: luxonResult.start ? luxonResult.start.setZone('Etc/UTC').toISO() : null,
+        end: luxonResult.end ? luxonResult.end.setZone('Etc/UTC').toISO() : null,
+      },
+    );
+  }
+
+  // This reserves the timezone offset, but still losing the timezone region
```

Comment:
I think we don't need the part "losing the timezone region" here. It is a little confusing to read in this context :v